File: java/1310. XOR Queries of a Subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        int[] xorpre = new int[arr.length+1];
        for (int i = 0; i < arr.length; i++) {
            xorpre[i+1] = xorpre[i] ^ arr[i];
        }
        for (int i = 0; i < queries.length; i++) {
            answer[i] = xorpre[queries[i][1]+1] ^ xorpre[queries[i][0]];
        }
        return answer;
    }
}

File: go/1310. XOR Queries of a Subarray.go
func xorQueries(arr []int, queries [][]int) []int {
	answer, xorpre := []int{}, []int{0}
	for i := 0; i < len(arr); i++ {
		xorpre = append(xorpre, xorpre[i]^arr[i])
	}
	for _, q := range queries {
		answer = append(answer, xorpre[q[1]+1]^xorpre[q[0]])
	}
	return answer
}